### PR TITLE
inject hooks on the first block after the main div wrapper

### LINKED ISSUE #8701: Single Product Template Compatibility Layer: On the first block, the hooks are injected before the div wrapper
## Describe the bug

On the first block, the hooks are injected before the div wrapper.

## To reproduce

1. Go to the Single Product Template.
2. Edit the template via code and insert [this markup](http://locale.local/wp-admin/site-editor.php?postType=wp_template&postId=woocommerce%2Fwoocommerce%2F%2Fsingle-product&canvas=edit) via code editor
3. Save.
4. Be sure that the Single Product Template is visible correctly.
5. Install the Code Snippets plugin.
6. Create a new snippet and copy-paste [this gist](https://gist.github.com/gigitux/b99b823b69dba7a615b82b4200ac5fbb).
7. Check that those hooks are put before the div wrapper:
- woocommerce_before_single_product
- woocommerce_before_single_product_summary
- woocommerce_single_product_summary

## Expected behavior

The hooks should inject after the **div** wrapper.

## Screenshots

image

### PR DESCRIPTION
**Pull request by gigitux**
During some tests, I noticed that the hooks on the first block are added before the `<div>` wrapper that we added to keep the compatibility with WC Core CSS.

image

This PR fixes this wrong behavior. 

Fixes #8701 

### Screenshots

| Before | After |
| ------ | ----- |
|image|image|

#### User Facing Testing

1. Go to the Single Product Template.
2. Edit the template via code and insert [this markup](https://gist.github.com/gigitux/e9323d6d280313f528996e34d62ae626) via the code editor
3. Save.
4. Be sure that the Single Product Template is visible correctly.
5. Install the Code Snippets plugin.
6. Create a new snippet and copy-paste [this gist](https://gist.github.com/gigitux/b99b823b69dba7a615b82b4200ac5fbb).
7. Check that those hooks are put AFTER the div wrapper:
- woocommerce_before_single_product
- woocommerce_before_single_product_summary
- woocommerce_single_product_summary
* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### CODE REVIEW

**Comment by gigitux** [comment]
Added as reviewer @Aljullu and @nefeline that reviewed the Compatibility Layer for the Single Product Template.

**Comment by nefeline** [comment]
hey @gigitux ! ref.:

> Edit the template via code and insert [this markup](http://locale.local/wp-admin/site-editor.php?postType=wp_template&postId=woocommerce%2Fwoocommerce%2F%2Fsingle-product&canvas=edit) via the code editor

Do you mind clarifying which markup is supposed to be inserted? The link is currently pointing to a local install. Thanks!

**Comment by gigitux** [comment]
> hey @gigitux ! ref.:
> 
> > Edit the template via code and insert [this markup](http://locale.local/wp-admin/site-editor.php?postType=wp_template&postId=woocommerce%2Fwoocommerce%2F%2Fsingle-product&canvas=edit) via the code editor
> 
> Do you mind clarifying which markup is supposed to be inserted? The link is currently pointing to a local install. Thanks!

Ops. I fixed the link! I created a GIF that shows how do this:

 

https://user-images.githubusercontent.com/4463174/224718658-3e506441-486f-4659-8353-fcff3f0c5d7a.mov

**Review by imanish003 — APPROVED** [review]
I tested the PR & it's working as expected 🚀

File: src/Templates/SingleProductTemplateCompatibility.php [hunk 1]
```diff
@@ -450,4 +448,27 @@ function( $carry, $block ) {
...
+
+	/**
+	 * Inject the hooks after the div wrapper.
+	 *
+	 * @param string $block_content Block Content.
+	 * @param array  $hooks Hooks to inject.
+	 * @return array
+	 */
+	private function inject_hooks_after_the_wrapper( $block_content, $hooks ) {
+		$closing_tag_position = strpos( $block_content, '>' );
```
Comment (imanish003) [inline src/Templates/SingleProductTemplateCompatibility.php:459]: I was just curious, Will there be always `>`? Do we always want to insert after first `>` character? 🤷🏻‍♂️

**Review by nefeline — APPROVED** [review]
Thanks for updating the instructions: nice work!

Test results:

>Go to the Single Product Template.
>Edit the template via code and insert [this markup](https://gist.github.com/gigitux/e9323d6d280313f528996e34d62ae626) via the code editor
>Save.
>Be sure that the Single Product Template is visible correctly.

✅ Confirmed the template is visible in the editor:

Screenshot 2023-03-14 at 10 27 33

✅  Confirmed the template is also properly displayed on the FE:

Screenshot 2023-03-14 at 10 29 01


> Create a new snippet and copy-paste [this gist](https://gist.github.com/gigitux/b99b823b69dba7a615b82b4200ac5fbb).
> Check that those hooks are put AFTER the div wrapper:

✅  Confirmed they are displayed after the wrapper div:

Screenshot 2023-03-14 at 10 26 27

File: src/Templates/SingleProductTemplateCompatibility.php (hunk 1 above)
Comment (gigitux) [reply src/Templates/SingleProductTemplateCompatibility.php:459]: > I was just curious, Will there be always >? 

Yes, this class adds this div:

```
<div class="wp-block-group woocommerce block ...">
```

> Do we always want to insert after first > character? 🤷🏻‍♂️

Yes! We want to inject the hooks after the close of the div tag. 

Does it make sense?

**Comment by gigitux** [comment]
Thanks for the review, folks! 

@imanish003 I will merge this PR, if you have some doubts feel free to continue the discussion in this PR!

File: src/Templates/SingleProductTemplateCompatibility.php (hunk 1 above)
Comment (imanish003) [reply src/Templates/SingleProductTemplateCompatibility.php:459]: That makes sense. Thanks for the explanation @gigitux. I just wanted to make sure that there will always be `>` character. 🤝

**Pull request merged**